AndreaAndreoli/senior-dev-tech-showcase
# Add Promotional Products Module with RabbitMQ and Elasticsearch Support

**Overview**:
This PR introduces the "Promotional Products" module for Magento 2, allowing admin management of promotional products with integrated RabbitMQ for async updates and Elasticsearch for search functionality.

**Key Features**:
- Added a new "Promotional Products" menu under the Marketing section.
- Admin grid with filtering, sorting, and mass actions (enable/disable promotions).
- Product promotion editing (start/end dates, discount percentage).
- RabbitMQ integration for processing promotion updates asynchronously.
- Elasticsearch integration for indexing promotional products and enabling search by promotion status.

**Documentation**:
A detailed requirements document is included in the module under `code/Demo/PromotionalProducts/PromotionalProducts_Requirements_Document.md` explaining all features, integrations, and testing instructions.

**Testing**:
1. Verify grid functionalities (pagination, filters, sorting) in the admin.
2. Test mass actions (enable/disable).
3. Edit promotion details and ensure changes are processed by RabbitMQ.
4. Verify that Elasticsearch indexes promotional products and supports searching by promotion status.
